legate/core/runtime.py

```diff
@@ -1543,6 +1539,193 @@ def create_store(
+
+        Returns
+        -------
+        AutoTask
+            A new automatically parallelized task
+
+        See Also
+        --------
+        Context.create_task
+        """
+
+        from .operation import AutoTask
+
+        # Check if the task id is valid for this library and the task
+        # has the right variant
+        machine = context.slice_machine_for_task(task_id)
+        unique_op_id = self.get_unique_op_id()
+        return AutoTask(context, task_id, unique_op_id, machine)
+
+    def create_copy(self) -> Copy:
+        """
+        Creates a copy operation.
+
+        Returns
+        -------
+        Copy
+            A new copy operation
+        """
+
+        from .operation import Copy
+
+        return Copy(
+            self.core_context,
+            self.get_unique_op_id(),
+            self.machine,
+        )
+
+    def issue_fill(
+        self,
+        lhs: Store,
+        value: Store,
+    ) -> None:
+        """
+        Fills the store with a constant value.
+
+        Parameters
+        ----------
+        lhs : Store
+            Store to fill
+
+        value : Store
+            Store holding the constant value to fill the ``lhs`` with
+
+        Raises
+        ------
+        ValueError
+            If the ``value`` is not scalar or the ``lhs`` is either unbound or
+            scalar
+        """
+        from .operation import Fill
+
+        fill = Fill(
+            self.core_context,
+            lhs,
+            value,
+            self.get_unique_op_id(),
+            self.machine,
+        )
+        fill.execute()
+
+    def tree_reduce(
+        self, context: Context, task_id: int, store: Store, radix: int = 4
+    ) -> Store:
+        """
+        Performs a user-defined reduction by building a tree of reduction
+        tasks. At each step, the reducer task gets up to ``radix`` input stores
+        and is supposed to produce outputs in a single unbound store.
+
+        Parameters
+        ----------
+        context : Context
+            Library to which the task id belongs
+
+        task_id : int
+            Id of the reducer task
+
+        store : Store
+            Store to perform reductions on
+
+        radix : int
+            Fan-in of each reducer task. If the store is partitioned into
+            :math:`N` sub-stores by the runtime, then the first level of
+            reduction tree has :math:`\\ceil{N / \\mathtt{radix}}` reducer
+            tasks.
+
+        Returns
+        -------
+        Store
+            Store that contains reduction results
+        """
+        from .operation import Reduce
+
+        if store.ndim > 1:
+            raise NotImplementedError(
+                "Tree reduction doesn't currently support "
+                "multi-dimensional stores"
+            )
+
+        result = self.create_store(store.type)
+
+        # A single Reduce operation is mapepd to a whole reduction tree
```

Comment:
No longer need to eagerly flush the scheduling widow (presumably for the partitions to be created)?

Reply:
yeah I don't think that's necessary anymore.